harness/determined/estimator/_estimator_trial.py

```diff
@@ -552,9 +561,13 @@ def _init_run_config(self, config: tf.estimator.RunConfig) -> tf.estimator.RunCo
             if session_config is None:
                 session_config = tf.compat.v1.ConfigProto()
             session_config.gpu_options.allow_growth = True
-            session_config.gpu_options.visible_device_list = str(
-                self.env.slot_ids[horovod.hvd.local_rank()]
-            )
+
+            if not self.env.experiment_config.get("data", {}).get(
```

Comment:
Why would we not want to just set this all of the time?

Reply:
added a comment